src/trading/appDataUtils.ts

```diff
@@ -11,15 +12,19 @@ export async function buildAppData(
   const quoteParams = { slippageBips: slippageBps }
   const orderClass = { orderClass: orderClassName }
 
-  const doc = await metadataApiSdk.generateAppDataDoc({
-    appCode,
-    metadata: {
-      quote: quoteParams,
-      orderClass,
-      partnerFee,
-    },
-    ...advancedParams,
-  })
+  const doc = await metadataApiSdk.generateAppDataDoc(
```

Comment:
Since `advancedParams` is a partial object, we should deep merge instead of shallow merge